File: Domain/Entities/UserSubscription.cs
namespace Domain.Entities
{
	public class UserSubscription
	{
		public int Id { get; set; }
		public User User { get; set; } = null!;
		public long UserId { get; set; }

		public Subscription Subscription { get; set; } = null!;
		public int SubscriptionId { get; set; }

		public DateTimeOffset ExpiresAt { get; set; }
		public DateTimeOffset BoughtAt { get; set; }

		public Guid? TransactionId { get; set; }

		public UserSubscriptionStatus Status { get; set; }
	}

	public enum UserSubscriptionStatus
	{
		Pending = 0,
		Completed = 1,
		Failed = 2
	}
}

File: Mobile/netflix/lib/domain/models/user_subscription.dart
class UserSubscription {
  final int id;
  final int userId;
  final int subscriptionId;
  final DateTime expiresAt;
  final DateTime boughtAt;
  final String? transactionId;
  final UserSubscriptionStatus status;

  UserSubscription({
    required this.id,
    required this.userId,
    required this.subscriptionId,
    required this.expiresAt,
    required this.boughtAt,
    required this.transactionId,
    required this.status,
  });
}

enum UserSubscriptionStatus { pending, completed, failed, cancelled }
